Parsers/ASimRegistryEvent/Parsers/vimRegistryEventVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,103 @@
+Parser:
+  Title: Registry Event ASIM Parser for VMware Carbon Black Cloud
+  Version: '0.1.0'
+  LastUpdated: Aug 08, 2023
+Product:
+  Name: VMware Carbon Black Cloud
+Normalization:
+  Schema: RegistryEvent
+  Version: '0.1.2'
+References:
+- Title: ASIM Registry Schema
+  Link: https://aka.ms/ASimRegistryEventDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: VMware Carbon Black Cloud documentation 
+  Link: https://developer.carbonblack.com/reference/carbon-black-cloud/
+        https://developer.carbonblack.com/reference/cb-threathunter/latest/event-search-fields/
+Description: |
+  This ASIM parser supports normalizing VMware Carbon Black Cloud logs to the ASIM Registry Event normalized schema. VMware Carbon Black Cloud events are captured through VMware Carbon Black Cloud data connector which ingests Carbon Black Audit, Notification and Event data into Microsoft Sentinel through the REST API.
+ParserName: vimRegistryEventVMwareCarbonBlackCloud
+EquivalentBuiltInParser: _Im_RegistryEvent_VMwareCarbonBlackCloud
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventTypeLookup = datatable (temp_action: string, EventType: string)
+      [
+      "ACTION_WRITE_VALUE", "RegistryValueSet",
+      "ACTION_CREATE_KEY", "RegistryKeyCreated",
+      "ACTION_DELETE_KEY", "RegistryKeyDeleted",
+      "ACTION_DELETE_VALUE", "RegistryValueDeleted",
+      "ACTION_RENAME_KEY", "RegistryKeyRenamed"
+  ];
+  let parser=(disabled: bool=false) {
+      CarbonBlackEvents_CL
+      | where not(disabled)
+            and eventType_s == "endpoint.event.regmod"
+            and isnotempty(regmod_name_s)
+      | extend temp_action = case(
+                          action_s has "|" and action_s has "delete",
+                          "ACTION_DELETE_KEY",
+                          action_s has "|" and action_s !has "delete",
+                          "ACTION_CREATE_KEY",
+                          action_s
+                      )
+      | where temp_action in ("ACTION_WRITE_VALUE", "ACTION_CREATE_KEY", "ACTION_DELETE_KEY", "ACTION_DELETE_VALUE", "ACTION_RENAME_KEY")
+      | lookup EventTypeLookup on temp_action
+      | extend
+          ActingProcessId = tostring(toint(process_pid_d)),
+          EventStartTime = todatetime(split(createTime_s, '+')[0]),
+          ParentProcessId = tostring(toint(parent_pid_d)),
+          AdditionalFields = bag_pack(
+                                "org_key", org_key_s,
+                                "process_guid", process_guid_s,
+                                "parent_guid", parent_guid_s 
```

Comment:
why we didn't map with these?
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/383b50c3-1353-48d5-a7f0-16817918ea36)

Reply:
In case of VMware Carbon Black Cloud, process_guid_s & parent_guid_s contains not only guid but combination of mulitple components like Org_Key, guid and some other values. Hence, we have added in AdditionalFields as we cannot add directly in Guid column due to invalid type.

Reply:
so we couldn't map ActingProcessGuid and ParentProcessGuid with process_guid and parent_guid ?

Reply:
No. as it is mentioned in document itself that this field not only contain guid but also combination of multiple components.
you can see the difference below:
Actual GUID format: "9a06dee6-06ef-44b3-9b6b-cb734c048f8b"
Carbon Black provided GUID: "DESJ9GN-003ddce3-00000270-00000000-1d90514c9d2519d" which is not in correct format.
![image](https://github.com/Azure/Azure-Sentinel/assets/107753570/e5fc3729-2ed0-41c8-8226-82f58844e98c)
Same apply for process_guid.

Hence we have added them into AdditionalFields.